distributed/worker.py

```diff
@@ -3259,8 +3267,7 @@ async def get_profile(
         return prof
 
     async def get_profile_metadata(self, comm=None, start=0, stop=None):
-        if stop is None:
-            add_recent = True
+        add_recent = stop is None
```

Comment:
Could you instead directly change to ``if stop is None:`` on line 3284?

Reply:
The line below this overwrites stop: `stop = stop or now`. I'm not clear on what these arguments do, but this would always make the condition true.